### PR TITLE
TDL-7301: Handle multiline `CRITICAL` error messages

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-7301: Handle multiline `CRITICAL` error messages
- Updated code to add `CRITICAL` for multiline error messages

# QA steps
 - [ ] automated tests passing
 - [ ] manual qa steps passing (list below)
 
# Risks

# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by savan-chovatiya — APPROVED** [review]

File: tap_salesforce/__init__.py [hunk 1]
```diff
@@ -417,5 +417,6 @@ def main():
         LOGGER.critical(e)
         sys.exit(1)
     except Exception as e:
-        LOGGER.critical(e)
```
Comment (dbshah1212) [inline tap_salesforce/__init__.py:420]: Will we get traceback after this change? In tap-facebook, they added Logger.exception specifically: https://github.com/singer-io/tap-facebook/blob/3318ad972a4c0df7b5f1c8bd57b942d5f6b39bf4/tap_facebook/__init__.py#L931

Comment (harshpatel4crest) [reply tap_salesforce/__init__.py:420]: Yes, we are getting traceback.

Comment (dbshah1212) [reply tap_salesforce/__init__.py:420]: As ask is to make it like a facebook, do we need to add Logger.exception specifically?

Comment (harshpatel4crest) [reply tap_salesforce/__init__.py:420]: I don't think so, the ask of this ticket was to Log `CRITICAL` for multiline error messages. And `LOGGER.exception` Logs `ERROR`.

**Review by dbshah1212 — APPROVED** [review]

**Review by luandy64 — APPROVED** [review]

**Review by RushT007 — APPROVED** [review]

**Pull request merged**